toonz/sources/toonz/mainwindow.cpp

```diff
@@ -800,151 +800,246 @@ void MainWindow::writeSettings() {
+  // StyleEditor
+  TPanel *styleEditorPane = TPanelFactory::createPanel(room, "StyleEditor");
+  if (styleEditorPane) {
+    styleEditorPane->setGeometry(QRect(34, 30, 259, 358));
+    geometries.push_back(styleEditorPane->geometry());
+    room->addDockWidget(styleEditorPane);
+    layout->dockItem(styleEditorPane);
   }
 
-  // Palette
-  TPanel *palettePane = TPanelFactory::createPanel(pltEditRoom, "LevelPalette");
-  if (palettePane) {
-    pltEditRoom->addDockWidget(palettePane);
-    layout->dockItem(palettePane, viewer, Region::bottom);
+  // Timeline
+  TPanel *timelinePane = TPanelFactory::createPanel(room, "Timeline");
+  if (timelinePane) {
+    timelinePane->setGeometry(QRect(34, 764, 1886, 211));
+    geometries.push_back(timelinePane->geometry());
+    room->addDockWidget(timelinePane);
+    layout->dockItem(timelinePane);
   }
 
-  // StyleEditor
-  TPanel *styleEditorPane =
-      TPanelFactory::createPanel(pltEditRoom, "StyleEditor");
-  if (styleEditorPane) {
-    pltEditRoom->addDockWidget(styleEditorPane);
-    layout->dockItem(styleEditorPane, viewer, Region::left);
+  // SceneViewer
+  TPanel *sceneViewerPane = TPanelFactory::createPanel(room, "SceneViewer");
+  if (sceneViewerPane) {
+    sceneViewerPane->setGeometry(QRect(297, 30, 1623, 730));
+    geometries.push_back(sceneViewerPane->geometry());
+    room->addDockWidget(sceneViewerPane);
+    layout->dockItem(sceneViewerPane);
   }
 
-  // Xsheet
-  TPanel *xsheetPane = TPanelFactory::createPanel(pltEditRoom, "Xsheet");
-  if (xsheetPane) {
-    pltEditRoom->addDockWidget(xsheetPane);
-    layout->dockItem(xsheetPane, palettePane, Region::left);
+  // ToolBar
+  TPanel *toolBarPane = TPanelFactory::createPanel(room, "ToolBar");
+  if (toolBarPane) {
+    toolBarPane->setGeometry(QRect(0, 30, 30, 945));
+    geometries.push_back(toolBarPane->geometry());
+    room->addDockWidget(toolBarPane);
+    layout->dockItem(toolBarPane);
   }
 
-  // Studio Palette
-  TPanel *studioPaletteViewer =
-      TPanelFactory::createPanel(pltEditRoom, "StudioPalette");
-  if (studioPaletteViewer) {
-    pltEditRoom->addDockWidget(studioPaletteViewer);
-    layout->dockItem(studioPaletteViewer, xsheetPane, Region::left);
+  // ToolOptions
+  TPanel *toolOptionsPane = TPanelFactory::createPanel(room, "ToolOptions");
+  if (toolOptionsPane) {
+    toolOptionsPane->setGeometry(QRect(0, 0, 1920, 26));
+    geometries.push_back(toolOptionsPane->geometry());
+    room->addDockWidget(toolOptionsPane);
+    layout->dockItem(toolOptionsPane);
   }
 
-  return pltEditRoom;
+  // LevelPalette
+  TPanel *levelPalettePane = TPanelFactory::createPanel(room, "LevelPalette");
+  if (levelPalettePane) {
+    levelPalettePane->setGeometry(QRect(34, 392, 259, 368));
+    geometries.push_back(levelPalettePane->geometry());
+    room->addDockWidget(levelPalettePane);
+    layout->dockItem(levelPalettePane);
+  }
+
+  DockLayout::State state(geometries, "-1 1 [ 4 [ 3 [ [ [ 0 5 ] 2 ] 1 ] ] ] ");
```

Comment:
I'm not questioning this one. . . just trying to learn.  What does this line do?

Reply:
Your guess is as good as mine. LOL.

Honestly I'm not sure entirely what it is.  I haven't really looked too deeply into it.  It some hierarchical notation of the regions the different panes sit in.